system-addon/lib/PingCentre.jsm

```diff
@@ -0,0 +1,149 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const {interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.importGlobalProperties(["fetch"]);
+
+XPCOMUtils.defineLazyModuleGetter(this, "ClientID",
+  "resource://gre/modules/ClientID.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "console",
+  "resource://gre/modules/Console.jsm");
+
+const PREF_BRANCH = "browser.ping-centre.telemetry.";
+
+const TELEMETRY_PREF = `${PREF_BRANCH}telemetry`;
```

Comment:
You could also create a pref branch to handle these in either of these two ways:

```js
const prefs = Services.prefs.getDefaultBranch(PREF_BRANCH);
// or
const prefs = new Preferences({branch: PREF_BRANCH});
```